### PR TITLE
minion: init at 3.0.12

### LINKED ISSUE #142868: Package request: Minion AddOn Manager (java version)
**Project description**
Premier AddOn Management 

**Metadata**

* homepage URL: https://minion.mmoui.com/
* source URL:  https://minion.mmoui.com/?download
* license: unfree
* platforms:  linux

### PR DESCRIPTION
**Pull request by PatrickDaG**
New package minion a MMO Addon manager for WOW and ESO from https://minion.mmoui.com/.
Close #142868

Adapted from @Vodurden, big thanks, and if you want to be credited somehow lmk.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Scrumplex** [comment]
```diff
diff --git a/pkgs/by-name/mi/minion/package.nix b/pkgs/by-name/mi/minion/package.nix
index 19cd5de8d478..a166e5d67679 100644
--- a/pkgs/by-name/mi/minion/package.nix
+++ b/pkgs/by-name/mi/minion/package.nix
@@ -2,23 +2,27 @@
   stdenvNoCC,
   lib,
   fetchzip,
-  openjdk22,
+  openjfx21,
+  openjdk21,
   makeDesktopItem,
-  javaPackages,
-  gsettings-desktop-schemas,
-  gtk3,
+  wrapGAppsHook3,
+  makeBinaryWrapper,
 }:
 
 let
-  openjfx = javaPackages.openjfx22.override { withWebKit = true; };
-  jdk = openjdk22.override (
-    prev:
-    prev
-    // {
-      enableJavaFX = true;
-      inherit openjfx;
-    }
-  );
+  openjfx = openjfx21.override { withWebKit = true; };
+  openjdk = openjdk21.override {
+    enableJavaFX = true;
+    inherit openjfx;
+  };
+
+  jvmArgs = [
+    "-cp $out/share/minion/lib"
+    "--add-exports=javafx.graphics/com.sun.javafx.css=ALL-UNNAMED"
+    "--add-exports=javafx.graphics/javafx.scene.image=ALL-UNNAMED"
+    "--add-opens=javafx.graphics/javafx.scene.image=ALL-UNNAMED"
+    "--add-opens=java.base/java.lang=ALL-UNNAMED"
+  ];
 in
 stdenvNoCC.mkDerivation rec {
   version = "3.0.12";
@@ -30,26 +34,24 @@ stdenvNoCC.mkDerivation rec {
     stripRoot = false;
   };
 
+  nativeBuildInputs = [
+    makeBinaryWrapper
+    wrapGAppsHook3
+  ];
+
+  dontWrapGApps = true;
+
   installPhase = ''
-    mkdir -p "$out/bin"
-    mkdir -p "$out/lib/${pname}"
-    cp ./Minion-jfx.jar "$out/lib/${pname}/Minion-jfx.jar"
-    cp -r ./lib "$out/lib/${pname}/lib"
+    runHook preInstall
 
-    cat > "$out/bin/${pname}" << EOF
-    #!${stdenvNoCC.shell}
-    CLASSPATH="$out/lib"
-    XDG_DATA_DIRS="${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name}:${gtk3}/share/gsettings-schemas/${gtk3.name}:XDG_DATA_DIRS"
-    exec ${lib.getExe jdk} \
-    --add-exports=javafx.graphics/com.sun.javafx.css=ALL-UNNAMED \
-    --add-exports=javafx.graphics/javafx.scene.image=ALL-UNNAMED \
-    --add-opens=javafx.graphics/javafx.scene.image=ALL-UNNAMED \
-    --add-opens=java.base/java.lang=ALL-UNNAMED \
-    -cp "$CLASSPATH ./lib" -jar "$out/lib/${pname}/Minion-jfx.jar" "\$@"
-    EOF
+    install -D Minion-jfx.jar "$out/share/minion/Minion-jfx.jar"
+    cp -r ./lib "$out/share/minion/"
 
-    chmod a+x "$out/bin/${pname}"
+    makeWrapper ${lib.getExe openjdk} $out/bin/minion \
+      "''${gappsWrapperArgs[@]}" \
+      --add-flags "${lib.concatStringsSep " " jvmArgs} -jar $out/share/minion/Minion-jfx.jar"
 
+    runHook postInstall
   '';
 
   desktopItems = [
@@ -67,7 +69,7 @@ stdenvNoCC.mkDerivation rec {
     homepage = "https://minion.mmoui.com/";
     license = lib.licenses.unfree;
     platforms = lib.platforms.all;
-    mainProgram = minion;
+    mainProgram = "minion";
     maintainers = with lib.maintainers; [ patrickdag ];
   };
 }
```

**Comment by Vodurden** [comment]
Thanks for taking this, I don't use my personal version anymore so I'm glad it was useful to someone else :slightly_smiling_face:

**Review by Scrumplex — APPROVED** [review]

File: pkgs/by-name/mi/minion/package.nix [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  makeDesktopItem,
+  wrapGAppsHook3,
+  makeBinaryWrapper,
+}:
+
+let
+  openjfx = openjfx21.override { withWebKit = true; };
+  openjdk = openjdk21.override {
+    enableJavaFX = true;
+    inherit openjfx;
```
Comment (Scrumplex) [inline pkgs/by-name/mi/minion/package.nix:16]: The JDK derivation was changed recently, and the override doesn't eval anymore.
```suggestion
    openjfx_jdk = openjfx;
```

**Review by FliegendeWurst — APPROVED** [review]
Builds on x64 Linux

**Pull request merged**